File: solutions/elixir/square-root/2/lib/square_root.ex
defmodule SquareRoot do
  @doc """
  Calculate the integer square root of a positive integer
  """
  @spec calculate(radicand :: pos_integer) :: pos_integer
  def calculate(radicand) do
    1..radicand
    |> Stream.take_while(fn n -> n * n <= radicand end)
    |> Enum.to_list()
    |> List.last()
  end
end

File: solutions/lua/square-root/2/square-root.lua
local SquareRoot = {}

function SquareRoot.square_root(radicand)
	local l = 0
	local a = 1
	local d = 3

	while a <= radicand do
		a = a + d
		d = d + 2
		l = l + 1
	end

	return l
end

return SquareRoot
